### PR TITLE
Add try-runtime to Trappist

### PR DESCRIPTION
**Pull request by brunopgalvao**
Inline with https://github.com/paritytech/trappist/issues/197

This PR adds try-runtime to the Trappist runtime.

Steps to run try-runtime on Trappist runtime:

1. Build node with try-runtime features
```
cargo build --release --features try-runtime
```
2. Make sure it worked
```
./target/release/trappist-node try-runtime --help
```

3. Copy the node and wasm
```
cp ./target/release/trappist-node node-try-runtime
```
```
cp ./target/release/wbuild/trappist-runtime/trappist_runtime.wasm runtime-try-runtime.wasm
```

4. Run a live network
```
zombienet-macos -p native spawn ./zombienet/trappist_rococo.toml
```

5. Run try-runtime against live state
```
./node-try-runtime \
    try-runtime \
    --runtime runtime-try-runtime.wasm \
    -lruntime=debug \
    on-runtime-upgrade \
    live --uri ws://127.0.0.1:9921
```

You should get output similar to this:
```
2023-06-16 10:58:17.934  INFO main remote-ext: since no at is provided, setting it to latest finalized head, 0xf1a17471551663f0e8378849afdff22d831544f13dcccf206affff3eeb435f70    
2023-06-16 10:58:17.935  INFO main remote-ext: since no prefix is filtered, the data for all pallets will be downloaded    
2023-06-16 10:58:17.951  INFO main remote-ext: scraping key-pairs from remote at block height 0xf1a17471551663f0e8378849afdff22d831544f13dcccf206affff3eeb435f70    
2023-06-16 10:58:17.957  INFO main remote-ext: Querying a total of 122 keys from prefix , splitting among 10 threads, 13 keys per thread    
2023-06-16 10:58:18.159  INFO main remote-ext: adding data for hashed prefix: , took 0s    
2023-06-16 10:58:18.159  INFO main remote-ext: adding data for hashed key: 3a636f6465    
2023-06-16 10:58:18.417  INFO main remote-ext: adding data for hashed key: 26aa394eea5630e07c48ae0c9558cef7f9cce9c888469bb1a0dceaa129672ef8    
2023-06-16 10:58:18.418  INFO main remote-ext: adding data for hashed key: 26aa394eea5630e07c48ae0c9558cef702a5c1b19ab7a04f536c519aca4983ac    
2023-06-16 10:58:18.444  INFO main remote-ext: initialized state externalities with storage root 0x75f665fb2b598a11ad9fec072ea85295aeb96fab12255115e8291e71d556a55b and state_version V1    
2023-06-16 10:58:19.284  INFO main try-runtime::cli: original spec: RuntimeString::Owned("trappist-rococo")-1, code hash: 9aa7befcf721885b7823c0b1139748c3fdd33eea093593420a6f6766804405f9    
2023-06-16 10:58:19.925  INFO main try-runtime::cli: new spec: RuntimeString::Owned("trappist-rococo")-1, code hash: 4d23ca9e9c67d3008e3ff57931d4c0dd7e8640506055baa4c3093cbd4e4e3962    
2023-06-16 10:58:22.104 DEBUG main runtime::frame-support: ✅ no migration for System    
2023-06-16 10:58:22.105  INFO main runtime::frame-support: ⚠️ ParachainSystem declares internal migrations (which *might* execute). On-chain `StorageVersion(2)` vs current storage version `StorageVersion(2)`    
2023-06-16 10:58:22.105 DEBUG main runtime::frame-support: ✅ no migration for RandomnessCollectiveFlip    
2023-06-16 10:58:22.106 DEBUG main runtime::frame-support: ✅ no migration for Timestamp    
2023-06-16 10:58:22.106 DEBUG main runtime::frame-support: ✅ no migration for ParachainInfo    
2023-06-16 10:58:22.106 DEBUG main runtime::frame-support: ✅ no migration for Balances    
2023-06-16 10:58:22.106 DEBUG main runtime::frame-support: ✅ no migration for TransactionPayment    
2023-06-16 10:58:22.106 DEBUG main runtime::frame-support: ✅ no migration for AssetTxPayment    
2023-06-16 10:58:22.106 DEBUG main runtime::frame-support: ✅ no migration for Authorship    
2023-06-16 10:58:22.106 DEBUG main runtime::frame-support: ✅ no migration for CollatorSelection    
2023-06-16 10:58:22.106 DEBUG main runtime::frame-support: ✅ no migration for Session    
2023-06-16 10:58:22.107 DEBUG main runtime::frame-support: ✅ no migration for Aura    
2023-06-16 10:58:22.107 DEBUG main runtime::frame-support: ✅ no migration for AuraExt    
2023-06-16 10:58:22.107  INFO main runtime::frame-support: ⚠️ XcmpQueue declares internal migrations (which *might* execute). On-chain `StorageVersion(2)` vs current storage version `StorageVersion(2)`    
2023-06-16 10:58:22.108  INFO main runtime::frame-support: ⚠️ PolkadotXcm declares internal migrations (which *might* execute). On-chain `StorageVersion(0)` vs current storage version `StorageVersion(0)`    
2023-06-16 10:58:22.108 DEBUG main runtime::frame-support: ✅ no migration for CumulusXcm    
2023-06-16 10:58:22.108  INFO main runtime::frame-support: ⚠️ DmpQueue declares internal migrations (which *might* execute). On-chain `StorageVersion(1)` vs current storage version `StorageVersion(1)`    
2023-06-16 10:58:22.109 DEBUG main runtime::frame-support: ✅ no migration for Contracts    
2023-06-16 10:58:22.109 DEBUG main runtime::frame-support: ✅ no migration for Assets    
2023-06-16 10:58:22.109 DEBUG main runtime::frame-support: ✅ no migration for Identity    
2023-06-16 10:58:22.109 DEBUG main runtime::frame-support: ✅ no migration for Uniques    
2023-06-16 10:58:22.109 DEBUG main runtime::frame-support: ✅ no migration for Scheduler    
2023-06-16 10:58:22.109 DEBUG main runtime::frame-support: ✅ no migration for Preimage    
2023-06-16 10:58:22.109 DEBUG main runtime::frame-support: ✅ no migration for LockdownMode    
2023-06-16 10:58:22.110 DEBUG main runtime::frame-support: ✅ no migration for Utility    
2023-06-16 10:58:22.110 DEBUG main runtime::frame-support: ✅ no migration for Multisig    
2023-06-16 10:58:22.110 DEBUG main runtime::frame-support: ✅ no migration for Council    
2023-06-16 10:58:22.110 DEBUG main runtime::frame-support: ✅ no migration for Treasury    
2023-06-16 10:58:22.110 DEBUG main runtime::frame-support: ✅ no migration for Democracy    
2023-06-16 10:58:22.110 DEBUG main runtime::frame-support: ✅ no migration for Sudo    
2023-06-16 10:58:22.110 DEBUG main runtime::frame-support: ✅ no migration for Dex    
2023-06-16 10:58:22.111 DEBUG main runtime::frame-support: ✅ no migration for AssetRegistry    
2023-06-16 10:58:22.125  INFO main try-runtime::cli: TryRuntime_on_runtime_upgrade executed without errors. Consumed weight = (300000000 ps, 0 byte), total weight = (500000000000 ps, 5242880 byte) (0.06 %, 0.00 %).  
```

### CODE REVIEW

File: runtime/stout/src/lib.rs [hunk 1]
```diff
@@ -779,18 +779,6 @@ impl_runtime_apis! {
 		}
 	}
 
-	#[cfg(feature = "try-runtime")]
```
Comment (kalaninja) [inline runtime/stout/src/lib.rs:782]: if you are removing conditional compilation from stout, it probably makes sense remove `try-runtime` feature from cargo.toml of stout.

Comment (brunopgalvao) [reply runtime/stout/src/lib.rs:782]: Done.

**Review by kalaninja — APPROVED** [review]

**Comment by mordamax** [comment]
bot help

**Comment by brunopgalvao** [comment]
bot try-runtime $ trappist

**Comment by mordamax** [comment]
bot try-runtime -v PIPELINE_SCRIPTS_REF=bruno-update-url $ trappist

**Comment by brunopgalvao** [comment]
bot try-runtime $ trappist

**Comment by brunopgalvao** [comment]
bot merge

**Pull request merged**